cql3/selection/selection.cc

```diff
@@ -577,8 +579,10 @@ void result_set_builder::flush_selectors() {
         // handled by process_current_row
         return;
     }
-    _result_set->add_row(_selectors->get_output_row());
-    _selectors->reset();
+    if (!_limit || _result_set->size() < *_limit) {
```

Comment:
This is better now, but I still don't get how does your patch short-cut fetching the data from the nodes once limit is reached? I see that this patch does produce a correct result set, but it doesn't stop the readers, since the result set builder will continue munching the input just will be sending it to /dev/null thanks to this change after generating LIMIT rows. But the goal is also to short-cut the execution and return the result set immediately.